NebulaPatcher/Patches/Transpilers/GameMain_Transpiler.cs

```diff
@@ -8,46 +8,55 @@ namespace NebulaPatcher.Patches.Transpiler
     {
-        //Ignore Pausing in the multiplayer:
+        //Enable Pausing only when Multiplayer.CanPause is true:
         //Change:  if (!this._paused)
-        //To:      if (!this._paused || Multiplayer.IsActive)
+        //To:      if (!(this._paused && Multiplayer.CanPause))
+        //Change:  if (this._fullscreenPaused && !this._fullscreenPausedUnlockOneFrame)
+        //To:      if (this._fullscreenPaused && Multiplayer.CanPause && !this._fullscreenPausedUnlockOneFrame)
 
         [HarmonyTranspiler]
         [HarmonyPatch(nameof(GameMain.FixedUpdate))]
-        static IEnumerable<CodeInstruction> PickupBeltItems_Transpiler(ILGenerator gen, IEnumerable<CodeInstruction> instructions)
+        static IEnumerable<CodeInstruction> FixedUpdate_Transpiler(IEnumerable<CodeInstruction> instructions, ILGenerator iL)
         {
-            var found = false;
-            var codes = new List<CodeInstruction>(instructions);
-            for (int i = 6; i < codes.Count; i++)
-            {
-                if (codes[i].opcode == OpCodes.Callvirt &&
-                    codes[i - 1].opcode == OpCodes.Ldc_I4_1 &&
-                    codes[i - 2].opcode == OpCodes.Br &&
-                    codes[i - 3].opcode == OpCodes.Ldc_I4_0 &&
-                    codes[i - 4].opcode == OpCodes.Br &&
-                    codes[i - 5].opcode == OpCodes.Ceq)
-                {
-                    found = true;
-                    //Define new jump for firct condition
-                    Label targetLabel = gen.DefineLabel();
-                    codes[i + 4].labels.Add(targetLabel);
+            var codeMatcher = new CodeMatcher(instructions, iL)
+                .MatchForward(true,
+                    new CodeMatch(OpCodes.Callvirt, AccessTools.Method(typeof(Player), "ApplyGamePauseState")),
+                    new CodeMatch(OpCodes.Ldarg_0),
+                    new CodeMatch(OpCodes.Ldfld, AccessTools.Field(typeof(GameMain), "_paused")),
+                    new CodeMatch(OpCodes.Brtrue)
+                );
 
-                    //Add my condition
-                    codes.InsertRange(i + 4, new CodeInstruction[] {
-                            new CodeInstruction(OpCodes.Call, AccessTools.Method(typeof(Multiplayer), "get_" + nameof(Multiplayer.IsActive))),
-                            new CodeInstruction(OpCodes.Brfalse_S, codes[i+3].operand)
-                    });
+            if (codeMatcher.IsInvalid)
+            {
+                NebulaModel.Logger.Log.Error("GameMain.FixedUpdate_Transpiler failed. Mod version not compatible with game version.");
+                return instructions;
+            }
 
-                    //Change jump of first condition
-                    codes[i + 3] = new CodeInstruction(OpCodes.Brfalse_S, targetLabel);
+            codeMatcher
+                .InsertAndAdvance(
+                    new CodeInstruction(OpCodes.Call, AccessTools.Method(typeof(Multiplayer), "get_" + nameof(Multiplayer.CanPause))),
```

Comment:
This should be `AccessTools.DeclaredPropertyGetter(typeof(Multiplayer), nameof(Multiplayer.CanPause))`

(though again this should probably be an option in the settings menu so -> MultiplayerOptions.PauseWhenNoConnections (idk what it should be called))

Reply:
Actually, I thought the same at first, but I don't think that we need to add this to the option menu since the only way to pause the game will be if you are in the game alone and in the ESC menu, I don't really think that there is a gain here to keep running the game in background if that's the case.